pilot/pkg/xds/ads.go

```diff
@@ -80,13 +81,10 @@ type Connection struct {
 	RouteConfigs map[string]*route.RouteConfiguration `json:"-"`
 	CDSClusters  []*cluster.Cluster
 
-	// Last nonce sent and ack'd (timestamps) used for debugging
-	ClusterNonceSent, ClusterNonceAcked   string
-	ListenerNonceSent, ListenerNonceAcked string
-	RouteNonceSent, RouteNonceAcked       string
-	RouteVersionInfoSent                  string
-	EndpointNonceSent, EndpointNonceAcked string
+	// Active contains the list of watched resources for the connection, keyed by the TypeUrl.
+	Active map[string]*model.WatchedResource
```

Comment:
I think this will likely need to be in the proxy not connection. Either that, or we move the TypeUrl part to proxy and keep this in connection. See https://github.com/istio/istio/pull/24877 for context

Reply:
Actually it probably makes sense to have it in the proxy. That way the configgen isn't doing something weird like looking at nonces.

Reply:
That makes sense. Actually proxy has Active already. I just moved the code to use it. And it will help to unify generators later